pinot-core/src/main/java/com/linkedin/pinot/core/query/scheduler/MultiLevelPriorityQueue.java

```diff
@@ -189,20 +186,20 @@ private SchedulerQueryContext takeNextInternal() {
       }
     }
 
-    SchedulerQueryContext query = null;
+    SchedulerQueryContext queryContext = null;
     if (currentWinnerGroup != null) {
-      ServerQueryRequest queryRequest = currentWinnerGroup.peekFirst().getQueryRequest();
+      queryContext = currentWinnerGroup.removeFirst();
+      ServerQueryRequest queryRequest = queryContext.getQueryRequest();
       if (LOGGER.isDebugEnabled()) {
-        sb.append(String.format(" Winner: %s: [%d,%d,%d,%d]", currentWinnerGroup.name(), queryRequest.getTimerContext().getQueryArrivalTimeMs(),
+        sb.append(String.format(" Winner: %s: [%d,%d,%d,%d]", currentWinnerGroup.name(), queryContext.getArrivalTimeNs(),
```

Comment:
I think we are looking to get absolute time in this log, ns time is not going to help us. I am not sure how to get that unless we record both System.currentMillis and System.nanos on query arrival. On the other hand, this is a debug log, so maybe we are ok ...

Reply:
Fixed.